compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -7573,7 +7573,7 @@ private List<BLangExpression> concatSimilarKindXMLNodes(List<BLangExpression> ex
         for (BLangExpression expr : exprs) {
             BType exprType;
             if (expr.getKind() == NodeKind.QUERY_EXPR) {
-                exprType = checkExpr(expr, xmlElementEnv, data.expType, data);
+                exprType = checkExpr(expr, xmlElementEnv, symTable.xmlType, data);
```

Comment:
Can we directly use `symTable.xmlType` here?
The following should also work?
```ballerina
    xml<xml:Element> docs = xml `<doc><num>1</num><num>2</num><num>3</num></doc>`;
    xml x3 = xml `<doc>${from xml:Element i in docs select i}</doc>`;
```

Reply:
Can use it directly right? So that the above sample will be equivalent to,
```bal
xml<xml:Element> docs = xml `<doc><num>1</num><num>2</num><num>3</num></doc>`;
xml x = from xml:Element i in docs select i;
xml x3 = xml `<doc>${x}</doc>`;
```
However, the above sample is not working due to issue #36333. Fixed with [0a23baa](https://github.com/ballerina-platform/ballerina-lang/pull/36287/commits/0a23baa9125d41e1014eac2be2a7f0a71d302ea6).